internal/functions/deploy/deploy.go

```diff
@@ -56,21 +67,87 @@ func getFunctionSlugs(fsys afero.Fs) ([]string, error) {
 	return slugs, nil
 }
 
-func bundleFunction(ctx context.Context, entrypointPath, importMapPath, buildScriptPath string) (*bytes.Buffer, error) {
-	denoPath, err := utils.GetDenoPath()
+func bundleFunction(ctx context.Context, dockerEntrypointPath, importMapPath string, fsys afero.Fs) (*bytes.Buffer, error) {
+	cwd, err := os.Getwd()
+	if err != nil {
+		return nil, err
+	}
+
+	// create temp directory to store generated eszip
+	tmpDir, err := os.MkdirTemp("", "eszip")
 	if err != nil {
 		return nil, err
 	}
-	// Bundle function and import_map with deno
-	args := []string{"run", "-A", buildScriptPath, entrypointPath, importMapPath}
-	cmd := exec.CommandContext(ctx, denoPath, args...)
-	var outBuf, errBuf bytes.Buffer
-	cmd.Stdout = &outBuf
-	cmd.Stderr = &errBuf
-	if err := cmd.Run(); err != nil {
-		return nil, fmt.Errorf("Error bundling function: %w\n%v", err, errBuf.String())
+	defer os.RemoveAll(tmpDir)
+	outputPath := "/root/eszips/output.eszip"
+
+	binds := []string{
+		// Reuse deno cache directory, ie. DENO_DIR, between container restarts
+		// https://denolib.gitbook.io/guide/advanced/deno_dir-code-fetch-and-cache
+		utils.EdgeRuntimeId + ":/root/.cache/deno:rw,z",
+		filepath.Join(cwd, utils.FunctionsDir) + ":" + dockerFuncDirPath + ":ro,z",
+		tmpDir + ":/root/eszips:rw,z",
```

Comment:
nitpick: might want to refactor as `dockerOutputDir` to reuse on L82

Reply:
Updated in 7c65bbe